src/rdkafka_metadata.c

```diff
@@ -1971,3 +1971,110 @@ rd_kafka_metadata_new_topic_with_partition_replicas_mock(int replication_factor,
         return rd_kafka_metadata_new_topic_mock(
             topics, topic_cnt, replication_factor, num_brokers);
 }
+
+/*
+ * @brief Handle update of metadata received in the produce or fetch tags.
+ *
+ * @param rk: used to look up current metadata.
+ * @param mdi: metadata parsed from the tags.
+ */
+void rd_kafka_metadata_update_op(rd_kafka_t *rk,
+                                 rd_kafka_metadata_internal_t *mdi) {
+        int i, j;
+        struct rd_kafka_metadata_cache_entry *rkmce;
+        rd_kafka_broker_t *rkb;
+        rd_bool_t cache_updated = rd_false;
+        rd_kafka_metadata_partition_t *mdp;
+        rd_kafka_metadata_partition_internal_t *mdp_int;
+
+
+        for (i = 0; i < mdi->metadata.broker_cnt; i++) {
+                /* TODO: How to fetch Proto? */
+                rd_kafka_broker_update(rk, RD_KAFKA_PROTO_PLAINTEXT,
+                                       &mdi->metadata.brokers[i], NULL);
+        }
+
+        for (i = 0; i < mdi->metadata.topic_cnt; i++) {
+                if (!RD_KAFKA_UUID_IS_ZERO(mdi->topics[i].topic_id)) {
```

Comment:
Add a variable here to use below too and 
```suggestion
                rd_bool_t by_id =
                    !RD_KAFKA_UUID_IS_ZERO(mdi->topics[i].topic_id);

                if (by_id) {
```

Reply:
Updated